src/lib/components/MyReservations.svelte

```diff
@@ -1,74 +1,59 @@
-<script>
-	import { datetimeToLocalDateStr, monthIdxFromDateStr } from '$lib/datetimeUtils';
+<script lang="ts">
+	import { datetimeToLocalDateStr } from '$lib/datetimeUtils';
 	import { minuteOfDay, beforeCancelCutoff } from '$lib/reservationTimes.js';
-	import { timeStrToMin, idx2month } from '$lib/datetimeUtils';
+	import { timeStrToMin } from '$lib/datetimeUtils';
 	import { user, userPastReservations, reservations } from '$lib/stores';
 	import { getContext } from 'svelte';
 	import Modal from './Modal.svelte';
 	import CancelDialog from './CancelDialog.svelte';
 	import RsvTabs from './RsvTabs.svelte';
 	import { Settings } from '$lib/settings.js';
 
-	export let resPeriod; /* past or upcoming */
+	import type { ReservationData, ReservationPeriod } from '$types';
+	import { ReservationCategory, ReservationStatus } from '$types';
+	import dayjs from 'dayjs';
 
-	function getResPeriod(rsv) {
+	export let resPeriod: ReservationPeriod = 'upcoming';
+
+	function getResPeriod(rsv: ReservationData) {
 		let view;
 		let today = new Date();
 		let todayStr = datetimeToLocalDateStr(today);
-		if (rsv.date > todayStr) {
+		if (rsv.date && rsv.date > todayStr) {
 			view = 'upcoming';
-		} else if (rsv.date < todayStr) {
+		} else if (rsv.date && rsv.date < todayStr) {
 			view = 'past';
 		} else {
-			let rsvMin;
-			if (['pool', 'classroom'].includes(rsv.category)) {
-				rsvMin = timeStrToMin(rsv.endTime);
-			} else if (rsv.category === 'openwater') {
+			let rsvMin: number = 0;
+			if (rsv.category === ReservationCategory.openwater) {
 				if (rsv.owTime === 'AM') {
 					rsvMin = timeStrToMin(Settings.get('openwaterAmEndTime', rsv.date));
 				} else if (rsv.owTime === 'PM') {
 					rsvMin = timeStrToMin(Settings.get('openwaterPmEndTime', rsv.date));
 				}
+			} else {
+				rsvMin = timeStrToMin(rsv.endTime);
 			}
-			view = rsvMin >= minuteOfDay(today) ? 'upcoming' : 'past';
+			view = rsvMin && rsvMin >= minuteOfDay(today) ? 'upcoming' : 'past';
 		}
 		return view;
 	}
 
-	const shortDate = (dateStr) => {
-		let re = /[0-9]+-([0-9]+)-[0]*([0-9]+)/;
-		let m = re.exec(dateStr);
-		let shortM = idx2month[parseInt(m[1]) - 1].slice(0, 3);
-		return m[2] + ' ' + shortM;
+	const bgColorByCategory = {
+		[ReservationCategory.pool]: 'from-pool-bg-from',
+		[ReservationCategory.openwater]: 'from-openwater-bg-from',
+		[ReservationCategory.classroom]: 'from-classroom-bg-from'
 	};
 
-	const bgColorFrom = (category) =>
-		category === 'pool'
-			? 'from-pool-bg-from'
-			: category === 'openwater'
-			? 'from-openwater-bg-from'
-			: category === 'classroom'
-			? 'from-classroom-bg-from'
-			: undefined;
-
-	const bgColorTo = (category) =>
```

Comment:
@getneil I think you removed the background color gradient (you kept from the 'from' color but removed the 'to' color).  Was that intentional?

Reply:
no. that was a mistake will fix that now.

Reply:
@nrxszvo this is fixed.